rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/dateTimePicker.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { mount } from "@vue/test-utils";
+import DateTimePicker from "../dateTimePicker.vue";
+import moment from "moment";
+const mountDateTimePicker = async (props = {}) => {
+  return mount(DateTimePicker, {
+    props: {
+      modelValue: moment().format(),
+      dateClass: "date-class",
+      timeClass: "time-class",
+      ...props,
+    },
+    global: {
+      stubs: {
+        "date-picker": true,
+        "time-picker": true,
+      },
+    },
+  });
+};
+describe("DateTimePicker.vue", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  it("emits update:modelValue when time changes", async () => {
+    const wrapper = await mountDateTimePicker();
+    const newTime = new Date();
+    const newTimeFormatted = moment(newTime).format();
+    const timePicker = wrapper.findComponent({ name: "time-picker" });
+    timePicker.vm.$emit("update:modelValue", newTime);
+    const emitted = wrapper.emitted("update:modelValue");
+    expect(emitted[0]).toEqual([newTimeFormatted]);
+  });
+  it("updates time and dateString when modelValue changes", async () => {
+    const wrapper = await mountDateTimePicker();
+    const newModelValue = moment().add(1, "days").format();
+    await wrapper.setProps({ modelValue: newModelValue });
+    expect(wrapper.vm.dateString).toBe(
+      moment(newModelValue).format("YYYY-MM-DD")
+    );
+    expect(wrapper.vm.time).toEqual(moment(newModelValue).toDate());
+  });
+  it("recalculates date when dateString changes", async () => {
+    const wrapper = await mountDateTimePicker();
+    const newDateString = moment().add(1, "days").format("YYYY-MM-DD");
+    await wrapper.setData({ dateString: newDateString });
+    expect(moment(wrapper.vm.time).format("YYYY-MM-DD")).toBe(newDateString);
+  });
+  it("binds v-model, class correctly and sets correct attributes on date-picker and time-picker", async () => {
+    const wrapper = await mountDateTimePicker();
+    const datePicker = wrapper.findComponent({ name: "date-picker" });
+    const timePicker = wrapper.findComponent({ name: "time-picker" });
+    // Check v-model and class for date-picker
+    expect(datePicker.attributes("modelvalue")).toBe(wrapper.vm.dateString);
+    expect(datePicker.attributes("class")).toContain(wrapper.vm.dateClass);
```

Comment:
suggest to change `wrapper.vm.dateClass` (both here and on the timepicker for the written class. It's more readable.